prow/plugins/approve/approve.go

```diff
@@ -171,6 +172,63 @@ func handleGenericComment(log *logrus.Entry, ghc githubClient, repo approvers.Re
 	)
 }
 
+// handleReviewEvent should only handle reviews that have no approval command.
+// Reviews with approval commands will be handled by handleGenericCommentEvent.
+func handleReviewEvent(pc plugins.PluginClient, re github.ReviewEvent) error {
+	botName, err := pc.GitHubClient.BotName()
+	if err != nil {
+		return err
+	}
+
+	opts := optionsForRepo(pc.PluginConfig, re.Repo.Owner.Login, re.Repo.Name)
+
+	// Check for an approval command is in the body. If one exists, let the
+	// genericCommentEventHandler handle this event. Approval commands override
+	// review state.
+	if approvalCommandMatcher(botName, opts.LgtmActsAsApprove, opts.ReviewActsAsApprove)(&comment{Body: re.Review.Body, Author: re.Review.User.Login}) {
```

Comment:
Is there an example of where we use the `approvalCommandMatcher()` later after the call? Why do we generate a matcher to use it immediately afterword in every case? Maybe we could add the comment and just return the `bool`?

Reply:
Yes. The returned matcher function is passed to `filterComments` in `handle()`. I refactored this to move the check logic into separate functions. Those functions can be used directly, as in this example, and the matcher function can OR the check functions.